### PR TITLE
T noleong/backup for migration

### PR DESCRIPTION
**Pull request by NokSamLeong**
add backup feature to automate databases backup for sql migration.

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: "Code Scanning - Action"
+
+on:
+  push:
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  CodeQL-Build:
+
```
Comment (brian-harris) [inline .github/workflows/codeql.yml:10]: is this file needed?  was it added by accident?

File: extensions/sql-migration/src/api/utils.ts [hunk 2]
```diff
@@ -865,3 +865,26 @@ export async function getBlobLastBackupFileNamesValues(lastFileNames: azureResou
...
+	credentials.rows.forEach((row) => {
+		credentialsValues.push({
+			name: row[0].displayValue,
+			displayName: row[0].displayValue
+		});
+	});
+	if (credentialsValues.length === 0) {
+		credentialsValues = [
+			{
+				displayName: 'no credential found',
```
Comment (brian-harris) [inline extensions/sql-migration/src/api/utils.ts:880]: please create resource strings for the new user strings by adding them to a constant in the 'strings.ts' file

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (NokSamLeong) [reply .github/workflows/codeql.yml]: removed

File: extensions/sql-migration/src/api/utils.ts (hunk 2 above)
Comment (NokSamLeong) [reply extensions/sql-migration/src/api/utils.ts]: pushed a new commit to fix

File: extensions/sql-migration/src/wizard/databaseBackupPage.ts [hunk 3]
```diff
@@ -1131,7 +1131,7 @@ export class DatabaseBackupPage extends MigrationWizardPage {
 							} else {
 								await this.disableBlobTableDropdowns(index, constants.BLOB_CONTAINER);
 							}
-							await this._blobContainerCredentialDropdown[index].updateProperty('enabled', value !== constants.NO_BLOBCONTAINERS_FOUND);
+							await this._blobContainerCredentialDropdowns[index].updateProperty('enabled', value !== constants.NO_BLOBCONTAINERS_FOUND);
```
Comment (brian-harris) [inline extensions/sql-migration/src/wizard/databaseBackupPage.ts:1134]: You can use the ".enabled" property directly

File: extensions/sql-migration/src/wizard/databaseBackupPage.ts [hunk 4]
```diff
@@ -1820,22 +1819,22 @@ export class DatabaseBackupPage extends MigrationWizardPage {
...
 					await this._networkShareBackupLoading[dbIndex].updateProperty('value', '');
 					await this._blobContainerBackupLoading[dbIndex].updateProperty('value', '');
 					break;
 				}
-				case 3: {
-					await this._networkShareBackupLoading[dbIndex].updateProperty('value', 'failed');
-					await this._blobContainerBackupLoading[dbIndex].updateProperty('value', 'failed');
+				case backupStatus.Failed: {
+					await this._networkShareBackupLoading[dbIndex].updateProperty('value', constants.StatusLookup['failed']);
+					await this._blobContainerBackupLoading[dbIndex].updateProperty('value', constants.StatusLookup['failed']);
```
Comment (brian-harris) [inline extensions/sql-migration/src/wizard/databaseBackupPage.ts:1837]: can 'failed' be replaced with the constant 'backupStatus.Failed'

Comment (NokSamLeong) [reply extensions/sql-migration/src/wizard/databaseBackupPage.ts:1837]: backupStatus.Failed is a number type

**Comment by Charles-Gagnon** [comment]
@brian-harris Is this PR still active?

**Comment by brian-harris** [comment]
this work is currently paused

**Comment by kburtram** [comment]
Based on above comments I'll close this.  Please reopen once work resumes.
